src/main/java/org/fogbowcloud/arrebol/execution/DockerTaskExecutor.java

```diff
@@ -1,146 +1,200 @@
+import org.fogbowcloud.arrebol.execution.dockerworker.WorkerDockerRequestHelper;
 import org.fogbowcloud.arrebol.models.command.Command;
 import org.fogbowcloud.arrebol.models.command.CommandState;
 import org.fogbowcloud.arrebol.models.task.Task;
-import org.fogbowcloud.arrebol.execution.exceptions.DockerStartException;
 import org.fogbowcloud.arrebol.models.task.TaskSpec;
 
+
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+
+import static java.lang.Thread.sleep;
 
 public class DockerTaskExecutor implements TaskExecutor {
 
     private String imageId;
     private String containerName;
 
-    private final String BASH = "/bin/bash";
-    private final String DOCKER_EXEC = "docker exec";
-    private final String DOCKER_RUN = "docker run --rm -idt --name";
-    private final String DOCKER_STOP = "docker stop";
 
+    private WorkerDockerRequestHelper workerDockerRequestHelper;
     private final Logger LOGGER = Logger.getLogger(DockerTaskExecutor.class);
 
-    public DockerTaskExecutor(String imageId, String containerName) {
+    public DockerTaskExecutor(String imageId, String containerName, String address) {
         this.imageId = imageId;
         this.containerName = containerName;
+        this.workerDockerRequestHelper = new WorkerDockerRequestHelper(address, containerName, imageId);
     }
 
     @Override
-    public TaskExecutionResult execute(Task task) {
-
+    public TaskExecutionResult execute(Task task){
         //FIXME: We should catch the errors when starting/finishing the container and move the task to its FAILURE state
         //FIXME: also, follow the SAME log format we used in the RawTaskExecutor
-
         TaskExecutionResult taskExecutionResult;
+
         Integer startStatus = this.start();
 
+        updateRequirements(task.getTaskSpec());
+
         if(startStatus != 0){
             LOGGER.error("Exit code from container start: " + startStatus);
-            throw new DockerStartException("Could not start container " + this.containerName);
+            throw new DockerStartException("Could not start container " + getContainerName());
         }
 
-        LOGGER.info("Successful started container " + this.containerName);
+        LOGGER.info("Successful started container " + getContainerName());
+        Command[] commands = getCommands(task);
+        LOGGER.info("Starting to execute commands [" + commands.length + "] of task " + task.getId());
+        int[] commandsResults = executeCommands(commands);
+
+        Integer stopStatus = this.stop();
+        if(stopStatus != 0){
+            LOGGER.error("Exit code from container stop: " + stopStatus);
+        }
 
+        taskExecutionResult = getTaskResult(commands, commandsResults);
+
+        LOGGER.info("Result of task [" + task.getId() + "]: " + taskExecutionResult.getResult().toString());
+        return taskExecutionResult;
+    }
+
+    private void updateRequirements(TaskSpec taskSpec){
```

Comment:
this method is odd. at first, I think it is very strange to have an update here. the executor should be stateless